File: scenarios/storage-python-account-mgmt/golden/storage_account_management.py
from __future__ import annotations

import os
import sys

from azure.core.exceptions import ClientAuthenticationError, HttpResponseError
from azure.identity import DefaultAzureCredential
from azure.mgmt.storage import StorageManagementClient
from azure.mgmt.storage.models import (
    BlobServiceProperties,
    Kind,
    Sku,
    SkuName,
    StorageAccountCreateParameters,
)


def run() -> None:
    subscription_id = os.environ["AZURE_SUBSCRIPTION_ID"]
    resource_group_name = os.environ["AZURE_RESOURCE_GROUP_NAME"]
    account_name = os.environ["AZURE_STORAGE_ACCOUNT_NAME"]
    location = os.environ.get("AZURE_LOCATION", "eastus")

    credential = DefaultAzureCredential()
    client = StorageManagementClient(credential, subscription_id)

    try:
        parameters = StorageAccountCreateParameters(
            sku=Sku(name=SkuName.STANDARD_LRS),
            kind=Kind.STORAGE_V2,
            location=location,
        )
        creation = client.storage_accounts.begin_create(
            resource_group_name,
            account_name,
            parameters,
        )
        creation.result()

        for account in client.storage_accounts.list_by_resource_group(
            resource_group_name,
        ):
            print(f"Storage account: {account.name}")

        properties = client.storage_accounts.get_properties(
            resource_group_name,
            account_name,
        )
        print(f"Storage account location: {properties.location}")

        blob_properties = client.blob_services.set_service_properties(
            resource_group_name,
            account_name,
            "default",
            BlobServiceProperties(is_versioning_enabled=True),
        )
        print(f"Blob versioning enabled: {blob_properties.is_versioning_enabled}")

        deletion = client.storage_accounts.begin_delete(
            resource_group_name,
            account_name,
        )
        deletion.result()
        print(f"Deleted storage account: {account_name}")
    except ClientAuthenticationError as error:
        print(f"Azure authentication failed: {error}", file=sys.stderr)
        raise
    except HttpResponseError as error:
        print(f"Storage management request failed: {error}", file=sys.stderr)
        raise


if __name__ == "__main__":
    run()

File: scenarios/storage-java-account-mgmt/golden/src/main/java/com/example/StorageAccountManagement.java
package com.example;

import com.azure.core.exception.HttpResponseException;
import com.azure.core.management.AzureEnvironment;
import com.azure.core.management.exception.ManagementException;
import com.azure.core.management.profile.AzureProfile;
import com.azure.identity.DefaultAzureCredential;
import com.azure.identity.DefaultAzureCredentialBuilder;
import com.azure.resourcemanager.storage.StorageManager;
import com.azure.resourcemanager.storage.models.BlobServiceProperties;
import com.azure.resourcemanager.storage.models.StorageAccount;
import com.azure.resourcemanager.storage.models.StorageAccountSkuType;

public final class StorageAccountManagement {
    private StorageAccountManagement() {
    }

    public static void main(String[] args) {
        String subscriptionId = requireEnvironment("AZURE_SUBSCRIPTION_ID");
        String resourceGroupName = requireEnvironment("RESOURCE_GROUP_NAME");
        String accountName =
                requireEnvironment("AZURE_STORAGE_ACCOUNT_NAME");

        DefaultAzureCredential credential =
                new DefaultAzureCredentialBuilder().build();
        AzureProfile profile = new AzureProfile(
                null, subscriptionId, AzureEnvironment.AZURE);
        StorageManager storageManager =
                StorageManager.authenticate(credential, profile);

        try {
            StorageAccount created = storageManager.storageAccounts()
                    .define(accountName)
                    .withRegion("eastus")
                    .withExistingResourceGroup(resourceGroupName)
                    .withSku(StorageAccountSkuType.STANDARD_LRS)
                    .withGeneralPurposeAccountKindV2()
                    .create();
            System.out.printf(
                    "Created storage account %s in %s with SKU %s%n",
                    created.name(),
                    created.regionName(),
                    created.skuType());

            for (StorageAccount account : storageManager.storageAccounts()
                    .listByResourceGroup(resourceGroupName)) {
                System.out.println("Storage account: " + account.name());
            }

            StorageAccount retrieved = storageManager.storageAccounts()
                    .getByResourceGroup(resourceGroupName, accountName);
            System.out.printf(
                    "Retrieved storage account: id=%s, region=%s, sku=%s, kind=%s%n",
                    retrieved.id(),
                    retrieved.regionName(),
                    retrieved.skuType(),
                    retrieved.kind());

            BlobServiceProperties blobService = storageManager.blobServices()
                    .getServicePropertiesAsync(resourceGroupName, accountName)
                    .block();
            BlobServiceProperties updatedBlobService = blobService.update()
                    .withBlobVersioningEnabled()
                    .apply();
            System.out.println(
                    "Blob versioning enabled: "
                            + updatedBlobService.isBlobVersioningEnabled());

            storageManager.storageAccounts()
                    .deleteByResourceGroup(resourceGroupName, accountName);
            System.out.println("Deleted storage account: " + accountName);
        } catch (ManagementException exception) {
            System.err.println(
                    "Storage management request failed: "
                            + exception.getMessage());
            throw exception;
        } catch (HttpResponseException exception) {
            System.err.println(
                    "Azure request failed: " + exception.getMessage());
            throw exception;
        }
    }

    private static String requireEnvironment(String name) {
        String value = System.getenv(name);
        if (value == null || value.isBlank()) {
            throw new IllegalStateException(
                    "Set " + name + " before running.");
        }
        return value;
    }
}
